os_net_config/impl_ifcfg.py

```diff
@@ -340,13 +342,13 @@ def iproute2_apply_commands(self, dev_name, filename, data):
             if changes['DOMAIN'] == 'added':
                 commands.append(f"echo 'search {new}' >> /etc/resolv.conf")
             elif changes['DOMAIN'] == 'modified':
-                commands.append(f"sed -i -e 's/{old}/{new}/g' "
-                                "/etc/resolv.conf")
+                commands.append(f"sed -i -e 's/search[ ]*{old}/search {new}/g'"
+                                " /etc/resolv.conf")
             elif changes['DOMAIN'] == 'removed':
-                commands.insert("sed -i '/search/d' /etc/resolv.conf")
-        if 'DNS1' in changes:
-            # Remove nameservers from /etc/resolv.conf
-            commands.append("sed -i '/nameserver/d' /etc/resolv.conf")
+                commands.insert("sed -i '/^search/d' /etc/resolv.conf")
+        if 'DNS1' in changes or 'DNS2' in changes:
+            # Remove all nameservers from /etc/resolv.conf
+            commands.append("sed -i '/^nameserver/d' /etc/resolv.conf")
```

Comment:
Can we use the same sed command pattern for DNS as well

Reply:
I don't believe so, I found edge cases where this didn't work. For example, let's say we have the following before update:
DNS1=10.0.1.14
DNS2=10.0.2.15
And the following after update:
DNS1=10.0.2.15
DNS2=10.0.3.16
When I ran the sed change to modify DNS1 in place, it would then appear like this:
DNS1=10.0.2.15
DNS2=10.0.2.15
When I then ran the update in place for DNS2 it would update both DNS1 and DNS2. I ran into other edge cases as well, and the only certain way of modifying both DNS1 and DNS2 was to remove both and add them back.

Reply:
I found some edge cases where this didn't work. For example if there were two DNS servers before update and one after. I suppose I could do this:
commands.append("sed -i '/^nameserver {old}/d' /etc/resolv.conf")

If DNS 2 was removed.

However I ran into some edge cases where there were issues. If DNS1 was moved to DNS2, and DNS1 was modified, I could run into a situation where /etc/resolv.conf looked like this:

ifcfg before change:
DNS1=10.0.1.14
DNS2=10.0.2.15
and after change:
DNS1=10.0.2.15
DNS2=10.0.3.16

Then after the nameserver for DNS1 was changed in place the /etc/resolv.conf would look like this:

nameserver 10.0.2.15
nameserver 10.0.2.15

So then if I ran "sed -i '^nameserver 10.0.2.15/d' /etc/resolv.conf" it would remove both nameservers and then add back only one, so DNS1 would not be added to the file and we would end up with only DNS2.

I found it was best to remove all nameservers and then immediately add the new ones. Notice that these commands are queued up immediately after one another:
sed -i '/^nameserver/d' /etc/resolv.conf
echo -e 'nameserver <ns1>' >> /etc/resolv.conf
echo -e 'nameserver <ns2> ' >> /etc/resolv.conf
So the chances of a nameserver lookup happening between the remove of both nameservers and the addition of the new nameservers is extremely small. Since this worked for all use cases I threw at it I didn't think it was worth it to risk an edge case hitting us if we were trying to modify in-place.